lib/builders/dataTypes/tiptapDataTypeBuilder.ts

```diff
@@ -0,0 +1,149 @@
+﻿import {DataTypeBuilder} from "./dataTypeBuilder";
+
+export class TiptapDataTypeBuilder extends DataTypeBuilder {
+  maxImageSize: number;
+  overlaySize: string;
+  dimensionsWidth: number;
+  dimensionsHeight: number;
+  ignoreUserStartNodes: boolean;
+  blocks: {contentElementTypeKey: string}[] = [];
+  mediaParentId: string;
+  extensions: string[];
+  toolbar: string[][][];
+
+  constructor() {
+    super();
+    this.editorAlias = "Umbraco.RichText";
+    this.editorUiAlias = "Umb.PropertyEditorUi.Tiptap";
+  }
+
+  withMaxImageSize(maxImageSize: number) {
+    this.maxImageSize = maxImageSize;
+    return this;
+  }
+
+  withDimensions(width: number, height: number) {
+    this.dimensionsWidth = width;
+    this.dimensionsHeight = height;
+    return this;
+  }
+
+  withOverlaySize(size: string) {
+    this.overlaySize = size;
+    return this;
+  }
+
+  withMediaFolderParentId(mediaParentId: string) {
+    this.mediaParentId = mediaParentId;
+    return this;
+  }
+
+  withIgnoreUserStartNodes(ignoreUserStartNodes: boolean) {
+    this.ignoreUserStartNodes = ignoreUserStartNodes;
+    return this;
+  }
+
+  addBlock(contentElementTypeKey: string) {
+    this.blocks.push({contentElementTypeKey});
+    return this;
+  }
+
+  addExtension(extension: string) {
+    this.extensions.push(extension);
+    return this;
+  }
+
+  addToolbar(toolbar: string[][]) {
+    this.toolbar.push(toolbar);
+    return this;
+  }
```

Comment:
Is there a reason why we don't have created separate builders for extensions and toolbar?

Reply:
I suppose we can add extensions and toolbar values directly since they are string values, so there is no need to create a separate builder for them.